airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/state/CheckpointManager.kt

```diff
@@ -140,24 +140,18 @@ abstract class StreamsCheckpointManager<T> : CheckpointManager<DestinationStream
     }
 
     override suspend fun flushReadyCheckpointMessages() {
-        if (!flushLock.tryLock()) {
-            log.info { "Flush already in progress, skipping" }
-            return
-        }
-        /*
-           Iterate over the checkpoints in order, evicting each that passes
-           the persistence check. If a checkpoint is not persisted, then
-           we can break the loop since the checkpoints are ordered. For global
-           checkpoints, all streams must be persisted up to the checkpoint.
-        */
-        try {
+        flushLock.withLock {
```

Comment:
oh, because the already-running flush attempt might not be operating on _all_ the available checkpoints?

Reply:
right. because this lock blocks the updater as well